pkg/ytconfig/base_generator.go

```diff
@@ -14,19 +14,22 @@ type BaseGenerator struct {
 	masterConnectionSpec   ytv1.MasterConnectionSpec
 	masterInstanceCount    int32
 	discoveryInstanceCount int32
+	masterCachesSpec       *ytv1.MasterCachesSpec
 }
 
 func NewRemoteBaseGenerator(
 	key types.NamespacedName,
 	clusterDomain string,
 	commonSpec ytv1.CommonSpec,
 	masterConnectionSpec ytv1.MasterConnectionSpec,
+	masterCachesSpec ytv1.MasterCachesSpec,
```

Comment:
It is inconsistent that one `New*` requires pointer and other is not.

Reply:
Fxied